### PR TITLE
UI changes for test plan

### PR DESCRIPTION
**Pull request by kaadhina**

### CODE REVIEW

File: Tasks/RunDistributedTests/task.json [hunk 1]
```diff
@@ -54,6 +54,51 @@
             "helpMarkDown": "Location where the test binaries have been dropped in the agent machine(s) as part of the 'Windows Machine File Copy' or 'Azure File Copy' task. System Environment Variables can also be used in location string. e.g., '%systemdrive%\\Tests', '%temp%\\DropLocation' etc.",
             "groupName": "setupOptions"
         },
+
+        {
+            "name": "testSelection",
```
Comment (acesiddhu) [inline Tasks/RunDistributedTests/task.json]: nit:
test type selection

File: Tasks/RunDistributedTests/task.json [hunk 2]
```diff
@@ -160,6 +205,27 @@
          "endpoint": "/$(system.teamProject)/_apis/vslabs/environments", 
          "selector": "jsonpath:$.value[*].name", 
          "authKey": "tfs:DevTestLabs" 
+     },
+     {
+         "target": "testPlan",
+         "endpoint":"/$(system.teamProject)/_apis/test/plans",
+         "selector":"jsonpath:$.value[*].name",
+         "keySelector":"jsonpath:$.value[*].id",
+         "authKey": "tfs:DevTestLabs"
```
Comment (acesiddhu) [inline Tasks/RunDistributedTests/task.json]: why do we need devtestlabs?

File: Tasks/RunDistributedTests/RunDistributedTests.ps1 [hunk 3]
```diff
@@ -46,18 +50,65 @@ Write-Verbose "UnregisterTestAgent script Path  = $unRegisterTestAgentLocation"
...
+
+$testConfigurationId = 0
+[int]::TryParse($testConfiguration, [ref]$testConfigurationId) 
+
 if($runTitleMemberExists)
 {
-    Invoke-RunDistributedTests -TestMachineGroup $testMachineGroup -SourceFilter $sourcefilters -TestCaseFilter $testFilterCriteria -RunSettingsPath $runSettingsFile -Platform $platform -Configuration $configuration -CodeCoverageEnabled $codeCoverageEnabled -TestRunParams $overrideRunParams -TestDropLocation $dropLocation -Connection $connection -TestConfiguration $testConfigurations -AutMachineGroup $autMachineGroup -UnregisterTestAgentScriptLocation $unregisterTestAgentScriptLocation -TestRunTitle $testRunTitle
+   if([string]::Equals($testSelection, "testPlan"))
+   {
+     if($testSelectionMemberExists)
```
Comment (acesiddhu) [inline Tasks/RunDistributedTests/RunDistributedTests.ps1]: if testplan is selected and agent is old we shuld fail at that time.

File: Tasks/RunDistributedTests/RunDistributedTests.ps1 [hunk 4]
```diff
@@ -10,11 +10,15 @@ param(
...
+    [string]$autMachineGroup,
+    [string]$testSelection,
+    [string]$testPlan,
+    [string]$testSuite,
+    [string]$testConfiguration
 )
 
 Function CmdletHasMember($memberName) {
-    $publishParameters = (gcm Publish-TestResults).Parameters.Keys.Contains($memberName) 
+    $publishParameters = (gcm Invoke-RunDistributedTests).Parameters.Keys.Contains($memberName)
```
Comment (acesiddhu) [inline Tasks/RunDistributedTests/RunDistributedTests.ps1]: shuld test compat, looks fine otherwise

Comment (kaadhina) [inline Tasks/RunDistributedTests/RunDistributedTests.ps1]: I've tested compat. Both old agent - new task, and other way around.

File: Tasks/RunDistributedTests/task.json (hunk 2 above)
Comment (kaadhina) [inline Tasks/RunDistributedTests/task.json]: The refresh button is tied to this auth key. we can add our own key and get rid of this. I'll take this as a bug fix.

**Pull request merged**